server/AccountManagement.java

```diff
@@ -1,48 +1,114 @@
+            if (!customer.checkExistingAccount(newAccountType)) {
+                Account.AccountType accountType;
+                if (newAccountType.equalsIgnoreCase(Account.AccountType.MAIN.toString())) {
+                    accountType = Account.AccountType.MAIN;
+                    customer.addAccount(new Account(accountType, 0.0));
+                    valid = true;
+                }
+                else if (newAccountType.equalsIgnoreCase(Account.AccountType.SAVINGS.toString())) {
+                    accountType = Account.AccountType.SAVINGS;
+                    customer.addAccount(new Account(accountType, 0.0));
+                    valid = true;
+                }
+                else if (newAccountType.equalsIgnoreCase(Account.AccountType.INVESTMENTS.toString())) {
+                    accountType = Account.AccountType.INVESTMENTS;
+                    customer.addAccount(new Account(accountType, 0.0));
+                    valid = true;
+                }
+                else {
+                    newBankClientHandler.sendOutput("An appropriate account type  must be selected [MAIN, SAVINGS, INVESTMENTS]. Please try again:");
+                }
+            }
+            else {
+                newBankClientHandler.sendOutput("The selected account type already exists. Please try again:");
+            }
+        }
+        newBankClientHandler.sendOutput("New account was successfully created. Exiting to Customer Menu.");
+        return newBankClientHandler.printCustomerMenu();
+    }
 
-            //ask for account name
-            newBankClientHandler.sendOutput("Please enter new account name ('q' to quit):");
-            newAccountName = newBankClientHandler.getInput();
+    public String removeAccount (Customer customer, NewBankClientHandler newBankClientHandler) {
 
-            //if user quits
-            if (checkQuitInput(newAccountName)){
-                return "MENU";
-            }
+        ArrayList<Account> existAccounts = customer.getAccounts();
+        Transaction transaction = new Transaction();
+        Account selectedAccount;
 
-            //if account name is blank, loop
-            if (newAccountName.trim().length()==0 || newAccountName.length() ==0){
-                newBankClientHandler.sendOutput("Account name must not be blank, please try again.");
-                continue;
+        newBankClientHandler.sendOutput("Selected: Remove existing account");
+        // Checking if the customer has more than one account types. Minimum one account per customer is needed.
+        if (existAccounts.size() < 2){
+            newBankClientHandler.sendOutput("You have only one account under your name. Request cannot be processed any further. Exiting to Customer Menu.");
+            return newBankClientHandler.printCustomerMenu();
+        }
+        else {
+            newBankClientHandler.sendOutput("Please enter name of your account you wish to remove or 'Q' to quit:");
```

Comment:
Instead of name it should be type with these new changes?

Reply:
Addressed.